### DISCUSSION TITLE
Errors in HA logs

### DISCUSSION THREAD

**Discussion by Captain0Kangaroo** [StealthChesnut/HA-FoxESS-Modbus#22] [General]
I'm getting these errors in my HA logs. 

Any ideas how to fix them?

Logger: homeassistant.components.template.template_entity
Source: components/template/template_entity.py:173 
Integration: Template (documentation, issues) 
First occurred: June 1, 2022, 23:34:34 (12 occurrences) 
Last logged: 11:59:37

TemplateError('ValueError: Template error: float got invalid input 'unavailable' when rendering template '{{ [0, ((states('sensor.pv1_power') | float + states('sensor.pv2_power') | float ) * 1)]|max }}' but no default was specified') while processing template 'Template("{{ [0, ((states('sensor.pv1_power') | float + states('sensor.pv2_power') | float ) * 1)]|max }}")' for attribute '_attr_native_value' in entity 'sensor.pv_total_power'
TemplateError('ValueError: Template error: float got invalid input 'unavailable' when rendering template '{% if (states('sensor.grid_ct') | float ) > 0 %} {{ states('sensor.grid_ct') | float * 1 }} {% else %} {{ (states('sensor.random_watts') | float - 5.5 ) / 1000 }} {% endif %}' but no default was specified') while processing template 'Template("{% if (states('sensor.grid_ct') | float ) > 0 %} {{ states('sensor.grid_ct') | float * 1 }} {% else %} {{ (states('sensor.random_watts') | float - 5.5 ) / 1000 }} {% endif %}")' for attribute '_attr_native_value' in entity 'sensor.feed_in_power'
TemplateError('ValueError: Template error: float got invalid input 'unavailable' when rendering template '{% if (states('sensor.grid_ct') | float ) < 0 %} {{ states('sensor.grid_ct') | float * -1 }} {% else %} {{ (states('sensor.random_watts') | float - 5.5 ) / 1000 }} {% endif %}' but no default was specified') while processing template 'Template("{% if (states('sensor.grid_ct') | float ) < 0 %} {{ states('sensor.grid_ct') | float * -1 }} {% else %} {{ (states('sensor.random_watts') | float - 5.5 ) / 1000 }} {% endif %}")' for attribute '_attr_native_value' in entity 'sensor.grid_consumption'
TemplateError('ValueError: Template error: float got invalid input 'unavailable' when rendering template '{% if (states('sensor.battery_discharge_power') | float ) > 0 %} {{ states('sensor.battery_discharge_power') | float * 1 }} {% else %} {{ (states('sensor.random_watts') | float - 5.5 ) / 1000 }} {% endif %}' but no default was specified') while processing template 'Template("{% if (states('sensor.battery_discharge_power') | float ) > 0 %} {{ states('sensor.battery_discharge_power') | float * 1 }} {% else %} {{ (states('sensor.random_watts') | float - 5.5 ) / 1000 }} {% endif %}")' for attribute '_attr_native_value' in entity 'sensor.battery_discharge'
TemplateError('ValueError: Template error: float got invalid input 'unavailable' when rendering template '{% if (states('sensor.battery_discharge_power') | float ) < 0 %} {{ states('sensor.battery_discharge_power') | float * -1 }} {% else %} {{ (states('sensor.random_watts') | float - 5.5 ) / 1000 }} {% endif %}' but no default was specified') while processing template 'Template("{% if (states('sensor.battery_discharge_power') | float ) < 0 %} {{ states('sensor.battery_discharge_power') | float * -1 }} {% else %} {{ (states('sensor.random_watts') | float - 5.5 ) / 1000 }} {% endif %}")' for attribute '_attr_native_value' in entity 'sensor.battery_charge'

**Comment by Daibutt** [comment]
What are the raw sensors that come from the modbus integration showing? (sensor.battery_discharge_power,  sensor.grid_ct, sensor.pv2_power etc..)
The logs suggest they are unavailable. 
Is your RS485>TCP converter online and has internet/wifi connection?

**Reply by Captain0Kangaroo** [reply]
It all seems to be working and it connected. No obvious errors that I can see.

![image](https://user-images.githubusercontent.com/96144685/171641005-2f6008bb-bd43-47bc-b36f-2ccf409dcd6f.png)

**Comment by Daibutt** [comment]
I'm not sure then. 
Is this transient, i.e. do your calcs for, say, sensor.pv_total_power work most of the time with an occasional drop out?
If so, I see this occasionally, usually coinciding with heavy network traffic (kids both streaming or gaming).
Just checked logs and I do get same log messages as you.

**Reply by Captain0Kangaroo** [reply]
Calculations all seem fine and I don't get any drop outs that I've observed.

Just wanted to to check and see if this is a problem or just ignore it.

**Reply by StealthChesnut** [reply]
Hey @Captain0Kangaroo, this is a side effect of a breaking change in the Templates feature in HomeAssistant.

I'll update the code to include a default value. There shouldn't be any problems, but once updated it'll make logs cleaner. I am also seeing the same in my logs :)
